pkg/cache/cache_test.go

```diff
@@ -69,185 +123,180 @@ var _ = Describe("Informer Cache", func() {
+			Expect(out).To(Equal(knownPod2))
+
+			By("altering a field in the retrieved pod")
+			*out.Spec.ActiveDeadlineSeconds = 4
+
+			By("verifying the pods are no longer equal")
+			Expect(out).NotTo(Equal(knownPod2))
+		})
+
+		It("should error out for missing objects", func() {
+			By("getting a service that does not exists")
+			svc := &kcorev1.Service{}
+			svcKey := client.ObjectKey{Namespace: "unknown", Name: "unknown"}
+
+			By("verifying that an error is returned")
+			err := informerCache.Get(context.Background(), svcKey, svc)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(Equal(true))
+		})
+	})
+
+	Describe("as an Informer", func() {
+		It("should be able to get informer for the object", func(done Done) {
+			By("getting a shared index informer for a pod")
+			pod := &kcorev1.Pod{
+				ObjectMeta: kmetav1.ObjectMeta{
+					Name:      "informer-obj",
+					Namespace: "default",
+				},
+				Spec: kcorev1.PodSpec{
+					Containers: []kcorev1.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+						},
+					},
+				},
+			}
+			sii, err := informerCache.GetInformer(pod)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sii).NotTo(BeNil())
+			Expect(sii.HasSynced()).To(Equal(true))
+
+			By("adding an event handler listening for object creation which sends the object to a channel")
+			out := make(chan interface{})
+			addFunc := func(obj interface{}) {
+				out <- obj
+			}
+			sii.AddEventHandler(kcache.ResourceEventHandlerFuncs{AddFunc: addFunc})
+
+			By("adding an object")
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl.Create(context.TODO(), pod)).To(Succeed())
+
+			By("verifying the object is received on the channel")
+			Eventually(out).Should(Receive(Equal(pod)))
+			close(done)
+		})
+
+		It("should be able to get an informer by group/version/kind", func(done Done) {
+			By("getting an shared index informer for gvk = core/v1/pod")
+			gvk := schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+			sii, err := informerCache.GetInformerForKind(gvk)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(sii).NotTo(BeNil())
+			Expect(sii.HasSynced()).To(Equal(true))
+
+			By("adding an event handler listening for object creation which sends the object to a channel")
+			out := make(chan interface{})
+			addFunc := func(obj interface{}) {
+				out <- obj
+			}
+			sii.AddEventHandler(kcache.ResourceEventHandlerFuncs{AddFunc: addFunc})
+
+			By("adding an object")
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			pod := &kcorev1.Pod{
+				ObjectMeta: kmetav1.ObjectMeta{
+					Name:      "informer-gvk",
+					Namespace: "default",
+				},
+				Spec: kcorev1.PodSpec{
+					Containers: []kcorev1.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+						},
+					},
+				},
+			}
+			Expect(cl.Create(context.TODO(), pod)).To(Succeed())
```

Comment:
`context.Background()`, not `context.TODO()`, unless you're really intending on coming back and changing this later (https://golang.org/pkg/context/#Background)